### PR TITLE
Fix picture link removal in "getAttachedData"

### PR DESCRIPTION
**Pull request by annando**
During further tests with the PR #11954 i detected some problems in the function `getAttachedData` that lead to leftover picture links. This is now fixed.

Also the whole "split" functionality is now tested with a lot of posts.

### CODE REVIEW

File: src/Content/Text/Plaintext.php [hunk 1]
```diff
@@ -299,10 +305,13 @@ private static function getParts(string $message, int $limit): array
 			if (strlen($part . $word) > ($limit - 8)) {
 				$parts[] = trim($part);
 				$part = '';
+				if (strlen($message) <= ($limit - 8)) {
+					$limit -= 23;
```
Comment (MrPetovan) [inline src/Content/Text/Plaintext.php]: Where is this `23` value coming from? Please indicate its source in a comment.

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Content/Text/Plaintext.php (hunk 1 above)
Comment (annando) [reply src/Content/Text/Plaintext.php]: I replaced it with a constant.

File: src/Content/Text/Plaintext.php [hunk 2]
```diff
@@ -23,9 +23,12 @@
 
 use Friendica\Core\Protocol;
 use Friendica\DI;
+use Friendica\Util\Network;
 
 class Plaintext
 {
+	const URL_LENGTH = 23;
```
Comment (MrPetovan) [inline src/Content/Text/Plaintext.php:31]: Thank you for adding a constant, can you please add documentation to it?

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Content/Text/Plaintext.php (hunk 2 above)
Comment (annando) [reply src/Content/Text/Plaintext.php:31]: Done

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**